### PR TITLE
Adding sync api support for cosmos encryption

### PR DESCRIPTION
**Pull request by simplynaveen20**
Adding sync api support for cosmos encryption.

Public surface changes in Cosmos project - 
1. Added sync CosmosClientEncryptionKey proxy object
2. Added CosmosClientEncryptionKey apis in CosmosDatabase

Public surface changes in Cosmos encryption project - 
1. Added CosmosEncryptionClient
2. Added CosmosEncryptionDatabase
3. Added CosmosEncryptionContainer

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    private final CosmosClient cosmosClient;
+
+    CosmosEncryptionClient(CosmosClient cosmosClient, EncryptionKeyStoreProvider encryptionKeyStoreProvider) {
+        this.encryptionKeyStoreProvider = encryptionKeyStoreProvider;
+        this.cosmosClient = cosmosClient;
+        this.cosmosAsyncClient = CosmosBridgeInternal.getCosmosAsyncClient(cosmosClient);
+        this.cosmosEncryptionAsyncClient = new CosmosEncryptionAsyncClient(cosmosAsyncClient, encryptionKeyStoreProvider);
+    }
+
+    public EncryptionKeyStoreProvider getEncryptionKeyStoreProvider() {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java:32]: does `EncryptionKeyStoreProvider` also expose sync api? or does it have any APIs with Mono?

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+     * Query for items in the current container using a {@link SqlQuerySpec}.
+     *
+     * @param <T>       the type parameter.
+     * @param query     the query.
+     * @param options   the query request options.
+     * @param classType the class type.
+     * @return a {@link CosmosPagedIterable}.
+     */
+    public <T> CosmosPagedIterable<T> queryItems(SqlQuerySpec query, CosmosQueryRequestOptions options,
+                                                 Class<T> classType) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java:142]: nit: in the other method signatures you have one arg per line. can we follow that here as well?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBridgeInternal.java [hunk 3]
```diff
@@ -32,6 +32,11 @@ public static AsyncDocumentClient getAsyncDocumentClient(CosmosClient client) {
         return client.asyncClient().getDocClientWrapper();
     }
 
+    @Warning(value = INTERNAL_USE_ONLY_WARNING)
+    public static CosmosAsyncClient getCosmosAsyncClient(CosmosClient client) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBridgeInternal.java:36]: we should avoid adding more BridgeInternal APIs as per discussion with the central SDK team.

You can rely on the pattern that we have here:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ImplementationBridgeHelpers.java

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ * synchronously
+ */
+@Beta(value = Beta.SinceVersion.V4_14_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class CosmosClientEncryptionKey {
+    private final CosmosDatabase database;
+    private String id;
+    private final CosmosAsyncClientEncryptionKey cosmosAsyncClientEncryptionKey;
+
+    CosmosClientEncryptionKey(String id, CosmosDatabase database,
+                                     CosmosAsyncClientEncryptionKey cosmosAsyncClientEncryptionKey) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java:20]: nit: indentation, and elsewhere on new APIs you have one arg per line. might be cleaner on the eye to follow that here as well.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java:35]: it expose sync apis

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * Create Cosmos Client with Encryption support for performing operations using client-side encryption.
+     *
+     * @param cosmosClient          Regular Cosmos Client.
+     * @param encryptionKeyStoreProvider encryptionKeyStoreProvider, provider that allows interaction with the master
+     *                                   keys.
+     * @return encryptionCosmosClient to perform operations supporting client-side encryption / decryption.
+     */
+    public static CosmosEncryptionClient createCosmosEncryptionClient(CosmosClient cosmosClient,
+                                                                                EncryptionKeyStoreProvider encryptionKeyStoreProvider) {
```
[the remark this answers is not in the document]
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java]: done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBridgeInternal.java (hunk 3 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBridgeInternal.java]: done, created CosmosClientAccessor

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java:143]: done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+package com.azure.cosmos;
+
+import com.azure.cosmos.models.CosmosClientEncryptionKeyResponse;
+import com.azure.cosmos.util.Beta;
+
+/**
+ * The type Cosmos clientEncryptionKey. This contains methods to operate on a cosmos clientEncryptionKey
+ * synchronously
+ */
+@Beta(value = Beta.SinceVersion.V4_14_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java:13]: Can you please update these Beta values to `V4_16_0` ?
You might need to add this `V4_16_0` to Beta.java enum.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        this.id = id;
+        this.cosmosAsyncClientEncryptionKey = cosmosAsyncClientEncryptionKey;
+    }
+
+    /**
+     * Get the id of the {@link CosmosClientEncryptionKey}
+     *
+     * @return the id of the {@link CosmosClientEncryptionKey}
+     */
+    @Beta(value = Beta.SinceVersion.V4_14_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java:32]: Same here, and other places, we need to update this to `V4_16_0`

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
Requesting small changes.

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+    /**
+     * Create Cosmos Client with Encryption support for performing operations using client-side encryption.
+     *
+     * @param cosmosClient               Regular Cosmos Client.
+     * @param encryptionKeyStoreProvider encryptionKeyStoreProvider, provider that allows interaction with the master
+     *                                   keys.
+     * @return encryptionCosmosClient to perform operations supporting client-side encryption / decryption.
+     */
+    public static CosmosEncryptionClient createCosmosEncryptionClient(CosmosClient cosmosClient,
```
[the remark this answers is not in the document]
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java:51]: This is inline as CosmosEncryptionAsyncClient createCosmosEncryptionAsyncClient(). Also we cannot restrict user if they want more client if they want for specific use case. Main cosmos sdk also allow use to create multiple client as needed

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    /**
+     * create item and encrypts the requested fields
+     *
+     * @param item           the Cosmos item represented as a POJO or Cosmos item object.
+     * @param partitionKey   the partition key.
+     * @param requestOptions request option
+     * @param <T>            serialization class type
+     * @return result
```
[the remark this answers is not in the document]
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionContainer.java]: done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java (hunk 7 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java]: done

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java (hunk 6 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientEncryptionKey.java]: done

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/CosmosEncryptionClient.java:51]: Makes sense!

**Review by kushagraThapar — APPROVED** [review]

**Pull request merged**